core/src/Cuda/Kokkos_Cuda_Instance.hpp

```diff
@@ -235,7 +235,7 @@ std::vector<Cuda> partition_space(const Cuda&, Args...) {
 }
 
 template <class T>
-std::vector<Cuda> partition_space(const Cuda&, std::vector<T>& weights) {
+std::vector<Cuda> partition_space(const Cuda&, std::vector<T> const& weights) {
```

Comment:
weights is not used anywhere here, just to get its size?

Reply:
Yes, for the device backends, we ignore the content of `weights`, we only care about the number of instances to create.

Reply:
can we add that as a comment so that it is clear for someone who reads it ?